test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs

```diff
@@ -390,5 +390,93 @@ public void AllowNoThrowBehaviorOnUnexpectedOptionAfterSubcommand()
             Assert.Equal(1, subCmd.RemainingArguments.Count);
             Assert.Equal(unexpectedOption, subCmd.RemainingArguments[0]);
         }
+
+        [Fact]
+        public void OptionsCanBeInherited()
+        {
+            var app = new CommandLineApplication();
+            var optionA = app.Option("-a|--option-a", "", CommandOptionType.SingleValue, inherited: true);
+            string optionAValue = null;
+
+            var optionB = app.Option("-b", "", CommandOptionType.SingleValue, inherited: false);
+
+            var subcmd = app.Command("subcmd", c =>
+            {
+                c.OnExecute(() =>
+                {
+                    optionAValue = optionA.Value();
+                    return 0;
+                });
+            });
+
+            Assert.Equal(2, app.GetOptions().Count());
+            Assert.Equal(1, subcmd.GetOptions().Count());
+
+            app.Execute("-a", "A1", "subcmd");
+            Assert.Equal("A1", optionAValue);
+
+            Assert.Throws<CommandParsingException>(() => app.Execute("subcmd", "-b", "B"));
+
+            Assert.Contains("-a|--option-a", subcmd.GetHelpText());
+        }
+
+        [Fact]
+        public void NestedOptionConflictThrows()
+        {
```

Comment:
Yes, when options are not inherited (default), only the subcommand-local options are searched. I'll add a sanity test for it.